ui/app/platforms/windows/WindowsStatusIcon.cc

```diff
@@ -347,8 +349,16 @@ WindowsStatusIcon::window_proc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lPara
           switch (lParam)
             {
             case WM_RBUTTONDOWN:
-              status_icon->show_menu();
+            {
+              bool taking = Core::get_instance()->is_taking();
```

Comment:
Could you use " status_icon->app->get_core()->is_taking();" instead of directly using the singleton Core class? (you can also remove the include ../../libs/core/src/Core.hh after this change)"

Reply:
Sure! I will also remove the test variables...
Btw, do I also need to add changes to MainWindow.cc in qt toolkit?
Do I also need to add changes to the equivalent of WindowsStatusIcon for Linux?

Reply:
The Qt port is still in development and only supported/compilable on macOS. You can add the same code there, but I can also do this at a later time when I continue with the Qt port.
On Linux, locking modes 'All' and 'Input' should prevent you from using the status icon menu. So I think no changes are required for Linux.

A recent change has caused a merge conflict BTW...

Reply:
@rcaelers Ah I see, I think I fixed the merge conflict.

Reply:
@rcaelers Seems pipeline still fails, I will check tonight.

Reply:
Fixed, I think.